circuit-construction/src/writer.rs

```diff
@@ -531,12 +617,16 @@ pub trait Cs<F: FftField + PrimeField> {
         acc
     }
 
+    /// Checks that a string of bits (with LSB first) correspond to the value inside variable `x`.
+    /// It splits the bitstring across rows, where each row takes care of 8 crumbs of 2 bits each.
+    ///
     fn assert_pack(&mut self, zero: Var<F>, x: Var<F>, bits_lsb: &[Var<F>]) {
         let crumbs_per_row = 8;
         let bits_per_row = 2 * crumbs_per_row;
         assert_eq!(bits_lsb.len() % bits_per_row, 0);
         let num_rows = bits_lsb.len() / bits_per_row;
 
+        // Reverse string of bits to have MSB first in the vector
         let bits_msb: Vec<_> = bits_lsb.iter().rev().collect();
 
         let mut a = self.var(|| F::from(2u64));
```

Comment:
Here, I don't see what `a` and `b` are being used for. They seem to be accumulating partially the crumb value, but they are never checked.